zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/migration/MigrateMessageEventSubprocessTest.java

```diff
@@ -0,0 +1,596 @@
+                    .userTask("userTask1")
+                    .endEvent("end")
+                    .done())
+            .withXmlResource(
+                Bpmn.createExecutableProcess(targetProcessId)
+                    .eventSubProcess(
+                        "sub2",
+                        s ->
+                            s.startEvent("start2")
+                                .message(m -> m.name("msg2").zeebeCorrelationKeyExpression("key"))
+                                .interrupting(false)
+                                .serviceTask("B", t -> t.zeebeJobType("task2"))
+                                .endEvent())
+                    .startEvent("start")
+                    .userTask("userTask2")
+                    .endEvent("end")
+                    .done())
+            .deploy();
+    final var processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(processId)
+            .withVariable("key", helper.getCorrelationValue())
+            .create();
+
+    ENGINE.message().withName("msg1").withCorrelationKey(helper.getCorrelationValue()).publish();
+    ENGINE.message().withName("msg1").withCorrelationKey(helper.getCorrelationValue()).publish();
+
+    Assertions.assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+                .withProcessInstanceKey(processInstanceKey)
+                .withElementId("A")
+                .limit(2))
+        .describedAs("Expect that the non-interrupting event subprocess is activated twice")
+        .hasSize(2);
+
+    final long targetProcessDefinitionKey =
+        extractProcessDefinitionKeyByProcessId(deployment, targetProcessId);
+
+    // when
+    ENGINE
+        .processInstance()
+        .withInstanceKey(processInstanceKey)
+        .migration()
+        .withTargetProcessDefinitionKey(targetProcessDefinitionKey)
+        .addMappingInstruction("userTask1", "userTask2")
+        .addMappingInstruction("sub1", "sub2")
+        .addMappingInstruction("A", "B")
+        .migrate();
+
+    // then
+    Assertions.assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_MIGRATED)
+                .withProcessInstanceKey(processInstanceKey)
+                .withElementType(BpmnElementType.EVENT_SUB_PROCESS)
+                .limit(2))
+        .hasSize(2)
+        .extracting(Record::getValue)
+        .describedAs("Expect that process definition key is changed")
+        .allMatch(v -> v.getProcessDefinitionKey() == targetProcessDefinitionKey)
+        .describedAs("Expect that bpmn process id and element id changed")
+        .allMatch(v -> v.getBpmnProcessId().equals(targetProcessId))
+        .allMatch(v -> v.getElementId().equals("sub2"))
+        .describedAs("Expect that version number did not change")
+        .allMatch(v -> v.getVersion() == 1);
+  }
+
+  @Test
+  public void shouldMigrateMessageEventSubprocess() {
```

Comment:
🔧This test verifies that the process instance's subscription to the message event-subprocess is migrated because of the start event mapping. Let's clarify this in the name of the test.

💭As this is a lot of information to cover, I prefer taking some shortcuts: mapped start event rather than mapped start event of event subprocess (this class is already focused on message event subprocess), and migrate subscription rather than migrate message subscription of process instance (it's not really relevant which element is subscribed to the event-subprocess)

```suggestion
  public void shouldMigrateSubscriptionForMappedStartEvent() {
```